app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -769,15 +764,24 @@ class BrowserFragment : Fragment(), BackHandler {
             }
             ToolbarMenu.Item.SaveToCollection -> showSaveToCollection()
             ToolbarMenu.Item.OpenInFenix -> {
-                // To not get a "Display Already Acquired" error we need to force remove the engineView here
-                swipeRefresh?.removeView(engineView as View)
-                pendingOpenInBrowserIntent = Intent(context, IntentReceiverActivity::class.java)
-                pendingOpenInBrowserIntent?.action = Intent.ACTION_VIEW
-                getSessionById()?.customTabConfig = null
+                // Release the session from this view so that it can immediately be rendered by a different view
+                engineView.release()
+
+                // Strip the CustomTabConfig to turn this Session into a regular tab and then select it
                 getSessionById()?.let {
+                    it.customTabConfig = null
                     requireComponents.core.sessionManager.select(it)
                 }
+
+                // Close this activity since it is no longer displaying any session
                 activity?.finish()
+
+                // Now switch to the actual browser which should now display our new selected session
+                val intent = Intent(context, IntentReceiverActivity::class.java).also {
+                    it.action = Intent.ACTION_VIEW
+                    it.flags = Intent.FLAG_ACTIVITY_NEW_TASK
```

Comment:
Added `NEW_TASK` here so that the original app task remains available to switch back to the other app - otherwise the browser may take that over.